src/react/helpers/hookStateFactory.ts

```diff
@@ -0,0 +1,28 @@
+import { useContext } from "react";
+import { SaleorContext } from "../context";
+import { DocumentNode, QueryResult, useQuery } from "@apollo/client";
+
+const useSaleorStateHook = (query: DocumentNode): QueryResult => {
+  const saleor = useContext(SaleorContext);
+
+  if (!saleor) {
+    throw new Error(
+      "Could not find saleor's apollo client in the context. Did you forget to wrap the root component in a <SaleorProvider>?"
+    );
+  }
+
+  const token = localStorage?.getItem("token");
+
+  return useQuery(query, {
+    client: saleor.client,
+    fetchPolicy: "cache-only",
```

Comment:
Why not `cache-and-network`?

Reply:
Because we want to rely only on Apollo cached data. If for some reason there are no data in the client's cache, the request for the data must be called again (most probably because of the logout, expired session) .